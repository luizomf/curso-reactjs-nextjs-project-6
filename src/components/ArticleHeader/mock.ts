import { ArticleHeaderProps } from '.';
import { data } from '../../api/dados.json';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
} as ArticleHeaderProps;
