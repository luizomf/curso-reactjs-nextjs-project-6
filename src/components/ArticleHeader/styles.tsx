import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
