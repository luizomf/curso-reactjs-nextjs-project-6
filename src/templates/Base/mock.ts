import { BaseTemplateProps } from '.';
import { data } from '../../api/dados.json';

export default {
  settings: data.setting,
  children: 'Ol√° mundo',
} as BaseTemplateProps;
