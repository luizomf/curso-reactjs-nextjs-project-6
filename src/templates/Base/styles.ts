import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  svg.search-ok-icon,
  svg.search-cancel-icon {
    width: 2.4rem;
    height: 2.4rem;
    margin-left: 1rem;
  }

  svg.search-cancel-icon {
    ${({ theme }) => css`
      color: ${theme.colors.secondary};
    `}
  }
`;

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xhuge};
    padding: 0 ${theme.spacings.large};
  `}
`;

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    padding: 0.5rem ${theme.spacings.small};

    &:disabled {
      outline: 0.1rem solid ${theme.colors.secondary};
    }
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.large};
  `}
`;

export const FooterContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
