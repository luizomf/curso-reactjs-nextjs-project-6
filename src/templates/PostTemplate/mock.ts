import { PostTemplateProps } from '.';
import { data } from '../../api/dados.json';

export default {
  settings: data.setting,
  post: data.posts[0],
} as PostTemplateProps;
