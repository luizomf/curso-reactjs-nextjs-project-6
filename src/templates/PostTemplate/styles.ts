import styled, { css } from 'styled-components';

export const TagsContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
  `}
`;
