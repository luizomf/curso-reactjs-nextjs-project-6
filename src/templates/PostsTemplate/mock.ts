import { PostsTemplateProps } from '.';
import { data } from '../../api/dados.json';

export default {
  settings: data.setting,
  posts: data.posts,
} as PostsTemplateProps;
