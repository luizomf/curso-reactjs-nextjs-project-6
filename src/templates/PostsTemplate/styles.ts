import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;

    &:disabled {
      background: ${theme.colors.darkerGray};
    }
  `}
`;
