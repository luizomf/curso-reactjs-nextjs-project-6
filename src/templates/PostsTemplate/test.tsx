import { renderTheme } from '../../styles/render-theme';
import { PostsTemplate, PostsTemplateProps } from '.';

import mock from './mock';

const props: PostsTemplateProps = mock;

describe('<PostsTemplate />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<PostsTemplate {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot without posts', () => {
    const { container } = renderTheme(
      <PostsTemplate {...props} posts={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
