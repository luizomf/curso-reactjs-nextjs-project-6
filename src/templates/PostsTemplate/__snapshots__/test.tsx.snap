// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsTemplate /> should match snapshot 1`] = `
.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c16 .c5 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c16 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c16:hover a {
  color: #dc143c;
}

.c16:hover img {
  opacity: 0.8;
}

.c17 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c14 {
  width: 100%;
}

.c15 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c20 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c20 p {
  margin: 2.4rem 0;
}

.c20 a,
.c20 a:visited,
.c20 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c20 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c20 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c20 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c20 pre code {
  color: inherit;
  background: inherit;
}

.c20 img {
  max-width: 100%;
}

.c20 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c20 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c20 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c20 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c20 ul,
.c20 ol {
  margin: 2.4rem 4.0rem;
}

.c20 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c20 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c20 table td,
.c20 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c20 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c19 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c19 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c21 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c9 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c18 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:360px) {
  .c15 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c20 {
    font-size: 2rem;
  }

  .c20 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: -0.5rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Otávio Miranda"
                src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="#target"
          target="_self"
        >
          O texto
        </a>
        <a
          class="c8"
          href="#link2"
          target="_blank"
        >
          o LINK 2
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Otávio Miranda - Uma descrição para o seu blog."
              src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            Otávio Miranda
          </h2>
          <p>
            Uma descrição para o seu blog.
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <a
              href="/post/vis-nobis-veritus-maluisset-id-nonumy-adversarium"
            >
              <img
                alt="Vis nobis veritus maluisset id, nonumy adversarium"
                class="c17"
                src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/vis-nobis-veritus-maluisset-id-nonumy-adversarium"
              >
                Vis nobis veritus maluisset id, nonumy adversarium
              </a>
            </h2>
            <p
              class=""
            >
              Fugit aliquip erroribus eu eam, an mel vidit doctus. Vis nobis veritus maluisset id, nonumy adversarium neglegentur est ut. Tation persius pri eu, modo minim fastidii mei no. Ad tota pertinax definiebas eam.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/sea-array-destructuring-dicat-fabulas-consetetur"
            >
              <img
                alt="Sea Array Destructuring dicat fabulas consetetur "
                class="c17"
                src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/sea-array-destructuring-dicat-fabulas-consetetur"
              >
                Sea Array Destructuring dicat fabulas consetetur 
              </a>
            </h2>
            <p
              class=""
            >
              Quod oblique mei ne, cetero delicatissimi sea cu, cu est harum dicit verterem. Tota forensibus ut usu, luptatum dignissim persecuti vel ex. Mea nostrud instructior cu, et referrentur theophrastus eum. Pro in iuvaret senserit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/quod-oblique-mei-ne-cetero-delicatissimi-sea"
            >
              <img
                alt="Quod oblique mei ne, cetero delicatissimi sea"
                class="c17"
                src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/quod-oblique-mei-ne-cetero-delicatissimi-sea"
              >
                Quod oblique mei ne, cetero delicatissimi sea
              </a>
            </h2>
            <p
              class=""
            >
              Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut. 
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c18"
    >
      <footer
        class="c19"
      >
        <div
          class="c20"
        >
          <p>
            O texto da footer
          </p>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c21"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`<PostsTemplate /> should match snapshot without posts 1`] = `
.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c14 {
  width: 100%;
}

.c15 {
  padding: 3.2rem;
  text-align: center;
  font-size: 1.6rem;
}

.c16 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c19 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c19 p {
  margin: 2.4rem 0;
}

.c19 a,
.c19 a:visited,
.c19 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c19 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c19 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c19 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c19 pre code {
  color: inherit;
  background: inherit;
}

.c19 img {
  max-width: 100%;
}

.c19 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c19 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c19 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c19 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c19 ul,
.c19 ol {
  margin: 2.4rem 4.0rem;
}

.c19 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c19 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c19 table td,
.c19 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c19 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c18 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c18 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c20 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c9 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c17 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:360px) {
  .c16 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c19 {
    font-size: 2rem;
  }

  .c19 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: -0.5rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Otávio Miranda"
                src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="#target"
          target="_self"
        >
          O texto
        </a>
        <a
          class="c8"
          href="#link2"
          target="_blank"
        >
          o LINK 2
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Otávio Miranda - Uma descrição para o seu blog."
              src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            Otávio Miranda
          </h2>
          <p>
            Uma descrição para o seu blog.
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <p
          class="c15"
        >
          Nenhum post encontrado aqui =(
        </p>
        <div
          class="c16"
        />
      </div>
    </div>
    <div
      class="c17"
    >
      <footer
        class="c18"
      >
        <div
          class="c19"
        >
          <p>
            O texto da footer
          </p>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c20"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
